File: c-notes/sorting algorithms/linear_search.c
#include <stdio.h>
#include <stdlib.h>

int linear_search(int *arr,int n,int element)
{
	for(int i=0;i<n;i++)
	{
		if(arr[i]==element)
			return i;
	}
	return -1;
}

void main()
{
	int n,x;
	printf("\n enter the number of elements in the array: ");
	scanf("%d",&n);int a[n];
	for(int i=0;i<n;i++)
	{
		printf("\n enter your element: ");
		scanf("%d",&a[i]);
	}
	printf("\n enter the element you want to search: ");
	scanf("%d",&x);
	int val=linear_search(a,n,x);
	if(val==-1)
		printf("\n element not in the list");
	else 
		printf("\n element %d is in the position %d",x,val);		
}

File: Java-notes/Searching-Algorithms/LinearSearch.java
/*
    Linear Search 

    *** output ***
    Element 2 is present at position 1 in array
*/

public class LinearSearch
{
    public static int linearSearch(int[] array,int target)
    {
        for(int i=0;i<array.length;i++)
        {
            if(array[i]==target)
            {
                return i;
            }
        }
        return -1;
    }
    public static void main(String[] args)
    {
        int[] array = {10,3,12,4,67,8,321,2};

        int index = linearSearch(array,3);

        if(index==-1)
            System.out.println("Element 2 is not found in the list");
        else 
            System.out.println("Element 2 is present at position "+index+" in array");
    }
}